core/experiments/assets/js/admin/behaviors/experiments-messages.js

```diff
@@ -149,29 +149,10 @@ export default class ExperimentsMessages {
 		} ).show();
 	}
 
-	joinDependenciesNames( array, glue = '', finalGlue = '' ) {
-		if ( '' === finalGlue ) {
-			return array.join( glue );
-		}
-
-		if ( ! array.length ) {
-			return '';
-		}
-
-		if ( 1 === array.length ) {
-			return array[ 0 ];
-		}
-
-		const clone = [ ...array ],
-			lastItem = clone.pop();
-
-		return clone.join( glue ) + finalGlue + lastItem;
-	}
-
 	showDependenciesDialog( experimentId ) {
 		const experiment = this.getExperimentData( experimentId ),
 			experimentName = experiment.title,
-			dialogMessage = this.joinDependenciesNames( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ), ', ' );
+			dialogMessage = new Intl.ListFormat( 'en' ).format( this.getExperimentDependencies( experimentId ).map( ( d ) => d.title ) );
```

Comment:
Hey @wittytzipporah,
Very nice investigation regarding the failing tests!
My only concern is with hard coding the `en` parameter. The word `and` for example might never be translated.
Maybe we should use `get_user_locale()` and expose it to the JS side. (But then we'll have to make sure the Locales from WP and this method matches).
WDYT?

Reply:
After some research into get_user_locale and JS Intl object, I think it's likely to be fine in all or nearly all cases. They are apparently both based on the BCP 47 standard. There may be some variance between browser implementations, but that is not likely to be in any of the major languages that Elementor is translated to.

Reply:
We also don't have to inject it ourselves, it's actually added in the <html> tag of the page by WP core - see wp-includes/template.php function _wp_admin_html_begin(). There don't appear to be any filters that we need to worry about removing it.